bluebutton-data-pipeline-app/src/test/java/gov/hhs/cms/bluebutton/datapipeline/app/S3ToFhirLoadAppIT.java

```diff
@@ -0,0 +1,425 @@
+
+		// Wait for it to exit with an error.
+		appProcess.waitFor(1, TimeUnit.MINUTES);
+
+		// Verify that the application exited as expected.
+		Assert.assertEquals(S3ToFhirLoadApp.EXIT_CODE_BAD_CONFIG, appProcess.exitValue());
+	}
+
+	/**
+	 * Verifies that {@link S3ToFhirLoadApp} exits as expected when asked to run
+	 * against an S3 bucket that doesn't exist. This test case isn't so much
+	 * needed to test that one specific failure case, but to instead verify that
+	 * the application dies as expected when something goes sideways.
+	 * 
+	 * @throws IOException
+	 *             (indicates a test error)
+	 * @throws InterruptedException
+	 *             (indicates a test error)
+	 */
+	@Test
+	public void missingBucket() throws IOException, InterruptedException {
+		Process appProcess = null;
+		try {
+			// Start the app.
+			ProcessBuilder appRunBuilder = createAppProcessBuilder(new Bucket("foo"));
+			appRunBuilder.redirectErrorStream(true);
+			appProcess = appRunBuilder.start();
+			new ProcessOutputConsumer(appProcess);
+
+			// Wait for it to exit with an error.
+			appProcess.waitFor(1, TimeUnit.MINUTES);
+
+			// Verify that the application exited as expected.
+			Assert.assertEquals(S3ToFhirLoadApp.EXIT_CODE_MONITOR_ERROR, appProcess.exitValue());
+		} finally {
+			if (appProcess != null)
+				appProcess.destroyForcibly();
+		}
+	}
+
+	/**
+	 * Verifies that {@link S3ToFhirLoadApp} works as expected when no data is
+	 * made available for it to process. Basically, it should just sit there and
+	 * wait for data, doing nothing.
+	 * 
+	 * @throws IOException
+	 *             (indicates a test error)
+	 * @throws InterruptedException
+	 *             (indicates a test error)
+	 */
+	@Test
+	public void noRifData() throws IOException, InterruptedException {
+		skipOnUnsupportedOs();
+
+		AmazonS3 s3Client = new AmazonS3Client(new DefaultAWSCredentialsProviderChain());
+		Bucket bucket = null;
+		Process appProcess = null;
+		try {
+			// Create the (empty) bucket to run against.
+			bucket = s3Client.createBucket(String.format("bb-test-%d", new Random().nextInt(1000)));
+
+			// Start the app.
+			ProcessBuilder appRunBuilder = createAppProcessBuilder(bucket);
+			appRunBuilder.redirectErrorStream(true);
+			appProcess = appRunBuilder.start();
+			ProcessOutputConsumer appRunConsumer = new ProcessOutputConsumer(appProcess);
+
+			// Wait for it to complete a scan.
+			Awaitility.await().atMost(Duration.ONE_MINUTE).until(() -> hasAScanFoundNothing(appRunConsumer));
+
+			// Stop the application.
+			sendSigterm(appProcess);
+			appProcess.waitFor(1, TimeUnit.MINUTES);
+
+			// Verify that the application exited as expected.
+			verifyExitValueMatchesSignal(SIGTERM, appProcess);
+		} finally {
+			if (appProcess != null)
+				appProcess.destroyForcibly();
+			if (bucket != null)
+				s3Client.deleteBucket(bucket.getName());
+		}
+	}
+
+	/**
+	 * Verifies that {@link S3ToFhirLoadApp} works as expected when no data is
```

Comment:
I think this comment was copied from the above test, but not updated for this test case